include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -678,7 +664,7 @@ struct __result_and_scratch_storage_impl : __result_and_scratch_storage_base
         else if (__supports_USM_device)
         {
             _T __tmp;
-            __exec.queue().memcpy(&__tmp, __scratch_buf.get() + __scratch_n + idx, 1 * sizeof(_T)).wait();
+            __q.memcpy(&__tmp, __scratch_buf.get() + __scratch_n + idx, 1 * sizeof(_T)).wait();
```

Comment:
I would propose to use 
`const_cast<__result_and_scratch_storage*>(this)->__q.memcpy(&__tmp, __scratch_buf.get() + __scratch_n + idx, 1 * sizeof(_T)).wait();` 
here... and keep `const` modificatory in the several other places, where the const was removed.

Reply:
One more variant - to rewrite this as
```C++
            const_cast<sycl::queue&>(__q)
                .memcpy(&__tmp, __scratch_buf.get() + __scratch_n + idx, 1 * sizeof(_T))
                .wait();
```
and restore `const` methods

Reply:
My point is that the methods 

1) `__get_value, __wait_and_get_value, __get_data` are data reading function and don't modify the instance of `__result_and_scratch_storage`. With that perspective the mentioned methods should be marked as const.

2) But, `sycl::queue::memcpy` method is defined as non-const according to the SYCl spec. Probably, it is because `memcpy` does submit a task to a queue and returns a sycl event. But it a kind of the SYCL implementation derails.

So, we have 3 ways:
1) To Remove "const" from  `__get_value, __wait_and_get_value, __get_data` as in this PR.
2) To make `__q` field as mutable.
3) To use `const_cast` just in the place where we really need it
`const_cast<queue&>(__q).memcpy(...)`

Let's discuss it.
I would vote for (3).

Reply:
@akukanov, @danhoeflinger, @mmichel11 what do you think about this?

Reply:
I don't have a very passionate opinion here, but unless we currently need `__get_value`, `__wait_and_get_value`, `__get_data` to be `const` functions, or unless we foresee a need for them to be I would just vote for (1).  

My reasoning being that I have a general preference against modifying constness via `mutable` / `const_cast` unless there is a very good reason to do so.

Reply:
So this looks like option (1) is our common decision and I save this code as is (without `const`-declared methods).
@MikeDvorskiy , @danhoeflinger, @mmichel11 could you please continue your reviews of this PR?

Reply:
My preference is for (2) (declare the queue as mutable) if the class has some other state and we want to ensure that state is not modified.

If the queue is the only field of the class, we might just go with (1) and say its methods are non-const.

However if `__result_and_scratch_storage` is in turn included into some other types as a field, and by making its methods non-const we would just "move" the issue to these other types, then I would again keep the methods const and change the queue to be mutable.